### PR TITLE
cob_light new features

### PR DESCRIPTION
**Pull request by benmaidel**
new features:
- pause / resume modes if mode with higher prio is requested / ended
- service / action returns a tracking id, with which the mode can be stopped
- new service `stopLightMode`
- changed command topic msg from `std_msgs/colorRGBA` to `cob_msgs/colorRGBAArray`
- separated mode types definition and mode message definition into separate messages
- new glow mode

### CODE REVIEW

File: cob_light/msg/LightModes.msg [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+uint8 BREATH =         3           # will change the LEDs smoothly with "frequency" from "color" to black
+uint8 BREATH_COLOR =   4           # will change the LEDs smoothly with "frequency" from "color" to black
+                                   # and flips color in time
+uint8 FADE_COLOR =     5           # will fade the colors in rainbow
+uint8 SEQ =            6           # executes one sequence after another as defined in sequences array
+uint8 CIRCLE_COLORS =  7           # circle through specific colors, if one color is set only one color will circle
+                                   # if more than one color is set, that this colors will circle through
+uint8 SWEEP =          8           # circle color from front to back on both sides
+uint8 DIST_APPROX =    9           # color led circle from green to red depended on the measured laserscan dists
+uint8 GLOW =          10           # glow the color
```
Comment (fmessmer) [inline cob_light/msg/LightModes.msg:13]: Why did you separate this from the `LightMode` message?
This is not a message that is meant to be sent on a topic....and thus confusing...

File: cob_light/ros/src/cob_light.cpp [hunk 2]
```diff
@@ -191,12 +195,12 @@ class LightControl
     //Subscribe to LightController Command Topic
     _sub = _nh.subscribe("command", 1, &LightControl::topicCallback, this);
 
-    //Subscribe to LightController Command Topic
-    _sub_mode = _nh.subscribe("light", 1, &LightControl::topicCallbackMode, this);
-
```
Comment (fmessmer) [inline cob_light/ros/src/cob_light.cpp:196]: why did you remove this subscriber?
what is/was the difference between `light` and `command`?

File: cob_light/common/include/glowColorMode.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ ****************************************************************/
+
+#ifndef GLOWCOLORMODE_H
+#define GLOWCOLORMODE_H
+
+#include <mode.h>
+
+class GlowColorMode : public Mode
```
Comment (fmessmer) [inline cob_light/common/include/glowColorMode.h:60]: just out of couriosity:
is there a documentation about the modes available....and - more importantly -  **what each mode is doing?** :wink:

File: cob_light/common/include/modeExecutor.h [hunk 4]
```diff
@@ -67,26 +68,31 @@ class ModeExecutor
...
-	void execute(cob_light::LightMode requestMode);
+	uint64_t execute(boost::shared_ptr<Mode> mode);
+	uint64_t execute(cob_light::LightMode requestMode);
 
 	int getExecutingPriority();
 	int getExecutingMode();
+	uint64_t getExecutingUId();
 
+	void pause();
+	void resume();
```
Comment (fmessmer) [inline cob_light/common/include/modeExecutor.h:79]: `pause` and `resume` is something the light driver handles internally only, right?
Depending on the modes that have been started and their respective priority...
It is not exposed to the user, correct?

File: cob_light/ros/src/cob_light.cpp [hunk 5]
```diff
@@ -191,12 +195,12 @@ class LightControl
...
     //Advertise light mode Service
     _srvServer = _nh.advertiseService("set_light", &LightControl::serviceCallback, this);
 
+    //Advertise stop mode service
+    _srvStopMode = _nh.advertiseService("stop_mode", &LightControl::stopMode, this);
```
Comment (fmessmer) [inline cob_light/ros/src/cob_light.cpp:202]: does a cancle of the `set_light` action do the same as a call to the `stop_light` service?

File: cob_light/msg/LightModes.msg (hunk 1 above)
Comment (benmaidel) [inline cob_light/msg/LightModes.msg:13]: I've separated them because we will not get md5 checksum error if we extend the LightModes with a new mode.

File: cob_light/ros/src/cob_light.cpp (hunk 2 above)
Comment (benmaidel) [inline cob_light/ros/src/cob_light.cpp:196]: because i've removed the possibility to switch modes via topic. This functionally was deprecated for a long time

File: cob_light/common/include/glowColorMode.h (hunk 3 above)
Comment (benmaidel) [inline cob_light/common/include/glowColorMode.h:60]: Nope, you can take a look inside the LightModes.msg. There you will find some explanations on the different modes

File: cob_light/common/include/modeExecutor.h (hunk 4 above)
Comment (benmaidel) [inline cob_light/common/include/modeExecutor.h:79]: correct. The user is only able to start new modes with a respective priority and stop running modes if he tracked them

File: cob_light/ros/src/cob_light.cpp (hunk 5 above)
Comment (benmaidel) [inline cob_light/ros/src/cob_light.cpp:202]: nope, because the action will return succeeded as soon as the requested mode started

**Comment by benmaidel** [comment]
successfully tested on raw3-3 and cob4-2

File: cob_light/ros/src/cob_light.cpp (hunk 5 above)
Comment (fmessmer) [inline cob_light/ros/src/cob_light.cpp:202]: so we  will not be able to call the stop functionality from a action client interface....just for the record...

**Comment by fmessmer** [comment]
FR ok
